analyzers/maven/maven.go

```diff
@@ -82,9 +82,9 @@ func Discover(dir string, options map[string]interface{}) ([]module.Module, erro
 				return nil
 			}
 
-			submodules, err := maven.Modules(path, dir, checked)
+			submodules, err := maven.Modules(filepath.Join(path, "pom.xml"), path, checked)
```

Comment:
Doing this because `files.Exists(path, "pom.xml")` is checked above (line 77).

Because this is just our attempt to _discover_ Maven modules, this is really all we can do. The official Maven documentation only ever speaks about `pom.xml` being the name of the main POM file in a project. All the real Maven projects I've seen have at least a _minimal_ `pom.xml` at the root of their project and a list of `<modules>` in it pointing to other POM files.